common/types/block_test.go

```diff
@@ -93,3 +93,33 @@ func TestBytesToNodeID(t *testing.T) {
 	_, err = BytesToNodeID(y[:])
 	r.Error(err, "Expected error converting too-long byte array to NodeID")
 }
+
+func TestLayerIDWraparound(t *testing.T) {
+	max := LayerID(^uint64(0))
```

Comment:
Consider using this for readability:

```suggestion
	max := LayerID(math.MaxUint64)
```